File: Project 4 :  server side js /controllers/userController.js
const { User } = require("../models/userModel");
const bcrypt = require("bcrypt");

const createUser = async (req, res) => {
  try {
    const { username, email, password } = req.body;

    if (!username || !email || !password) {
      return res.status(400).json({ message: "All details of user Required!" });
    }

    const existingUser = await User.findOne({ email });

    // console.log(existingUser)

    if (existingUser) {
      return res.status(400).json({ message: "User Already Exists!" });
    }

    const passHash = await bcrypt.hash(password, 10);

    if (passHash) {
      const newuser = await User.create({
        username,
        email,
        password: passHash,
      }); // db query

      if (newuser) {
        res.status(201).json({ message: "User Created Succesfully!" });
      }
    }
  } catch (error) {
    console.log(error);
  }
};

module.exports = { createUser };

File: Project 10: Ef core mvc/WebApplication1/Controllers/UserController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using WebApplication1.Data;
using WebApplication1.Models;

namespace WebApplication1.Controllers
{
    public class UserController : Controller
    {
        private readonly SqlDbContext _dbContext;

        public UserController(SqlDbContext dbContext)
        {
            _dbContext = dbContext;
        }

        public string ErrorMessage = "";
        public string SuccessMessage = "";


        [HttpGet]
        public IActionResult Register()
        {
            return View();
        }

        [HttpGet]
        public IActionResult Login()
        {
            return View();
        }

        [HttpGet]
        public IActionResult Dashboard()
        {
            return View();
        }



        [HttpPost]
        public async Task<ActionResult> Register(User user)
        {
            try
            {
                // var finduser = await _dbContext.Users.FindAsync(user.Email);    this method works with {PK}
                var findUser = await _dbContext.Users.FirstOrDefaultAsync(u => u.Email == user.Email);
                if (findUser != null)
                {
                    ViewData["ErrorMessage"] = "User already exists.";
                    return View();
                }

                user.Password = BCrypt.Net.BCrypt.HashPassword(user.Password);

                if (ModelState.IsValid)
                {
                    await _dbContext.Users.AddAsync(user);
                    await _dbContext.SaveChangesAsync();
                    ViewData["SuccessMessage"] = "User created successfully!";
                    return View();
                }
                else
                {
                    ViewData["ErrorMessage"] = "All feilds are required!";
                    return View();
                }
            }
            catch (Exception ex)
            {
                ViewData["ErrorMessage"] = "An error occurred: ";
                Console.WriteLine(ex.Message);
                return View();
            }
        }





        [HttpPost]
        public async Task<IActionResult> Login(User user)
        {
            try
            {
                var findUser = await _dbContext.Users.FirstOrDefaultAsync(u => u.Email == user.Email);

                if (findUser == null || !BCrypt.Net.BCrypt.Verify(user.Password, findUser.Password))
                {
                    ViewData["ErrorMessage"] = "Invalid email or password.";
                    return View();
                }

                ViewData["SuccessMessage"] = "Login successful!";
                return RedirectToAction("Dashboard");
            }
            catch (Exception ex)
            {
                ViewData["ErrorMessage"] = "An error occurred: " + ex.Message;
                return View();
            }
        }


    }









}
